Plugins/Public/base_plugin/PlayerCommands.cpp

```diff
@@ -451,6 +451,119 @@ namespace PlayerCommands
 		PrintUserCmdText(client, L"OK");
 	}
 
+	bool CheckForBase(PlayerBase *base, uint client)
+	{
+		if (!base)
+		{
+			PrintUserCmdText(client, L"ERR Not in player base");
+			return true;
+		}
+
+		if (!clients[client].admin)
+		{
+			PrintUserCmdText(client, L"ERR Access denied");
+			return true;
+		}
+		return false;
+	}
+
+	void BaseAddAllyFac(uint client, const wstring &args)
+	{
+		PlayerBase *base = GetPlayerBaseForClient(client);
+		if (CheckForBase(base, client)) return;
+
+		uint tag = 0;
+		try
+		{
+			tag = std::stoi(GetParam(args, ' ', 2));
+		}
+		catch (exception)
+		{
+			PrintUserCmdText(client, L"ERR No tag");
+			return;
+		}
+
+		if ((base->ally_factions).find(tag) != (base->ally_factions).end())
+		{
+			PrintUserCmdText(client, L"ERR Tag already exists");
+			return;
+		}
+
+		wstring theaffiliation = (HkGetWStringFromIDS(Reputation::get_name(tag))).c_str();
+		if (theaffiliation == L"Object Unknown")
+		{
+			PrintUserCmdText(client, L"ERR Undefined faction");
+			return;
+		}
+		base->ally_factions.insert(tag);
+
+		base->Save();
+		PrintUserCmdText(client, L"OK");
+	}
+
+	void BaseClearAllyFac(uint client, const wstring &args)
+	{
+		PlayerBase *base = GetPlayerBaseForClient(client);
+		if (CheckForBase(base, client)) return;
+		base->ally_factions.clear();
+		base->Save();
+		PrintUserCmdText(client, L"OK");
+	}
+
+	void BaseRmAllyFac(uint client, const wstring &args)
+	{
+		PlayerBase *base = GetPlayerBaseForClient(client);
+		if (CheckForBase(base, client)) return;
+
+		uint tag = 0;
+		try
+		{
+			tag = std::stoi(GetParam(args, ' ', 2));
+		}
+		catch (exception)
+		{
+			PrintUserCmdText(client, L"ERR No tag");
+			return;
+		}
+
+		if ((base->ally_factions).find(tag) == (base->ally_factions).end())
+		{
+			PrintUserCmdText(client, L"ERR Tag does not exist");
+			return;
+		}
+
+		base->ally_factions.erase(tag);
+		base->Save();
+		PrintUserCmdText(client, L"OK");
+	}
+
+	void BaseLstAllyFac(uint client, const wstring &cmd)
+	{
+		PlayerBase *base = GetPlayerBaseForClient(client);
+		if (CheckForBase(base, client)) return;
+
+		set<uint>::iterator it;
+		for (it = base->ally_factions.begin(); it != base->ally_factions.end(); ++it)
+		{
+			wstring theaffiliation = (HkGetWStringFromIDS(Reputation::get_name(*it))).c_str();
+			std::wstring widestr = std::wstring(theaffiliation.begin(), theaffiliation.end());
+			const wchar_t* widecstr = widestr.c_str();
+			PrintUserCmdText(client, L"%d, %s", *it, widecstr);
+		}
+		PrintUserCmdText(client, L"OK");
+	}
+	void BaseViewMyFac(uint client, const wstring &cmd)
+	{
+		uint aff = GetAffliationFromClient(client);
+
+		wstring theaffiliation = (HkGetWStringFromIDS(Reputation::get_name(aff))).c_str();

```

Comment:
extra brackets

Reply:
After searching for a bit, I think that redundant brackets (parenthesis) does nothing for the resulting code
https://www.quora.com/Extra-parentheses-is-have-any-effect-on-C-C++-program-or-a-time-of-compilation
Except for some unique situations.

Fixed anyway to -> wstring theaffiliation = HkGetWStringFromIDS(Reputation::get_name(aff)).c_str();